extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/NarayanaJtaRecorder.java

```diff
@@ -32,34 +31,115 @@ public void setNodeName(final TransactionManagerConfiguration transactions) {
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.SubordinationManagerXAResourceOrphanFilter"));
+    }
+
+    private void setRecoveryModuleClassNames(RecoveryEnvironmentBean recoveryEnvironmentBean) {
+        recoveryEnvironmentBean.setRecoveryModuleClassNames(Arrays.asList(
+                "com.arjuna.ats.internal.arjuna.recovery.AtomicActionRecoveryModule",
+                "com.arjuna.ats.internal.txoj.recovery.TORecoveryModule",
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.SubordinateAtomicActionRecoveryModule",
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.XARecoveryModule",
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.CommitMarkableResourceRecordRecoveryModule"));
+    }
+
+    private void setExpiryScannerClassNames(RecoveryEnvironmentBean recoveryEnvironmentBean) {
+        recoveryEnvironmentBean.setExpiryScannerClassNames(Collections.singletonList(
+                "com.arjuna.ats.internal.arjuna.recovery.ExpiredTransactionStatusManagerScanner"));
+    }
+
+    private void setRecoveryPort(RecoveryEnvironmentBean recoveryEnvironmentBean) {
+        recoveryEnvironmentBean.setRecoveryPort(4712);
+    }
+
+    private void setRecoveryListener(RecoveryEnvironmentBean recoveryEnvironmentBean) {
+        recoveryEnvironmentBean.setRecoveryListener(true);
+    }
+
+    public void setConfig(TransactionManagerConfiguration transactions) {
+        Properties systemProperties;
+        final SecurityManager sm = System.getSecurityManager();
+
+        if (sm != null) {
+            systemProperties = AccessController.doPrivileged((PrivilegedAction<Properties>) System::getProperties);
+        } else {
+            systemProperties = System.getProperties();
+        }
+
+        // directly set the Narayana property beans to avoid the need to parse property files
+        if (transactions.checkForNarayanaSystemProperties) {
+            RecoveryEnvironmentBean recoveryEnvironmentBean;
+            JTAEnvironmentBean jtaEnvironmentBean;
+
+            BeanPopulator.getDefaultInstance(CoordinatorEnvironmentBean.class, systemProperties);
+            BeanPopulator.getDefaultInstance(CoreEnvironmentBean.class, systemProperties);
+            BeanPopulator.getDefaultInstance(ObjectStoreEnvironmentBean.class, systemProperties);
+
+            recoveryEnvironmentBean = BeanPopulator.getDefaultInstance(RecoveryEnvironmentBean.class, systemProperties);
+            jtaEnvironmentBean = BeanPopulator.getDefaultInstance(JTAEnvironmentBean.class, systemProperties);
+
+            // set defaults if not set via system properties
+            if (!systemProperties.containsKey("JTAEnvironmentBean.xaResourceOrphanFilterClassNames")) {
+                setXaResourceOrphanFilterClassNames(jtaEnvironmentBean);
+            }
+
+            if (!systemProperties.containsKey("RecoveryEnvironmentBean.recoveryModuleClassNames")) {
```

Comment:
in Narayana property model it's possible to define the system property in this shortened way of `RecoveryEnvironmentBean.recoveryModuleClassNames` and the other valid property naming is with class name `com.arjuna.ats.arjuna.common.RecoveryEnvironmentBean.recoveryModuleClassNames`
(plus for some of the properties there are defined aliases as e.g. for this one I assume it's valid `com.arjuna.ats.arjuna.recovery.recoveryExtension`)

Reply:
`NameEnvironmentBean.propertyName` is the preferred format. The other format is legacy.

The goal of allowing system properties is to provide a mechanism to configure an option that isn't exposed in the quarkus config model. It is not the intention to make everything configurable via the quarkus model (it is similar to WildFly where we still allow a back door to enable access to certain Narayana functionality).

Reply:
@mmusgrov ok, I see. Have we stated the preferred format somewhere?

Reply:
It is in our docs

Reply:
@ochaloup Yes we have (both in the javadoc for this PR and in the narayana documentation).

Reply:
@ochaloup yes we have stated it. It is in the javadoc for this PR and it is in the Narayana documentation.